### PR TITLE
Skip test_max_reads_limits_fusion on XPU (SYCL TLA precision issue on PVC)

### LINKED ISSUE #181699: DISABLED test_max_reads_limits_fusion (__main__.OverFusionTest)
Platforms: xpu

This test was disabled because it is failing on main branch ([recent examples](https://torch-ci.com/failure?failureCaptures=%5B%22test%2Finductor%2Ftest_mix_order_reduction.py%3A%3AOverFusionTest%3A%3Atest_max_reads_limits_fusion%22%5D)).

cc @mruberry @chauhang @penguinwu @voznesenskym @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @ipiszy @kadeng @muchulee8 @amjames @aakhundov @coconutruben @jataylo @gujinghui @fengyuan14

### PR DESCRIPTION
**Pull request by ZhaoqiongZ**
## Summary

Adds `@skipIfXpu` to `OverFusionTest.test_max_reads_limits_fusion` in `test/inductor/test_mix_order_reduction.py` to replace the current bot-disabled state (pytorch/pytorch#181699) with an explicit, documented skip.

## Root Cause

This is a **training** test (forward + backward + grad check) that uses `F.scaled_dot_product_attention` . The precision failure is confirmed to be caused by the SDPA backward path with sycl tla building with LTS driver.

On XPU, `F.scaled_dot_product_attention` dispatches via the `OVERRIDEABLE` backend (SDPBackend=4) to the XPU Flash Attention implementation, which uses **SYCL TLA** for gradient accumulation in the backward pass. SYCL TLA on PVC with LTS driver does not currently guarantee precision, causing the compiled gradient to exceed the `tol=5e-2` threshold.

```
F.scaled_dot_product_attention(q, k, v, is_causal=True)
  -> _fused_sdp_choice() = 4 (OVERRIDEABLE)
  -> aten._scaled_dot_product_flash_attention.default
  -> _scaled_dot_product_fused_attention_overrideable_xpu  (mkldnn/xpu/Attention.cpp)
```

**Ablation confirmation**: replacing the SDPA call in the UT with `SDPBackend.MATH` (all other kernels unchanged — RMSNorm backward, MLP backward, mix_order_reduction triton kernels) makes the test **pass**, even with the CI Docker environment (Ubuntu 24.04 Noble) and CI build wheels. This isolates the root cause exclusively to the SDPA backward path.

The skip will be removed once oneDNN adds training (backward) support for this Flash Attention pattern on PVC, routing the SDPA dispatch through oneDNN instead of the SYCL TLA path.

Tracking: https://github.com/intel/torch-xpu-ops/issues/4094

## Changes

- `test/inductor/test_mix_order_reduction.py`: add `@skipIfXpu` with full explanation and tracking issue link to `test_max_reads_limits_fusion`

## Test Plan

```bash
ZE_AFFINITY_MASK=0 python test/inductor/test_mix_order_reduction.py \
  OverFusionTest.test_max_reads_limits_fusion -v
# Result: OK (skipped=1)
```

Closes #181699

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @ipiszy @kadeng @muchulee8 @amjames @chauhang @aakhundov @coconutruben @jataylo

### CODE REVIEW

**Comment by jansel** [comment]
@claude explain those CI failures, check if they are related to this PR, and review the changes

**Review by etaf — APPROVED** [review]

File: test/inductor/test_mix_order_reduction.py [hunk 1]
```diff
@@ -1161,6 +1161,12 @@ class OverFusionTest(TestBase):
     regression. See #179423.
     """
 
+    @skipIfXpu(
+        msg="XPU Flash Attention backward uses SYCL TLA which does not guarantee "
+        "precision on PVC; fails on Ubuntu 24.04 LTS2 (CI) builds with MKL 2024.2. "
+        "Will be re-enabled once oneDNN adds training support for this attention "
+        "pattern on PVC. See https://github.com/intel/torch-xpu-ops/issues/4094"
```
Comment (guangyey) [inline test/inductor/test_mix_order_reduction.py]:
```suggestion
        msg="XPU selects Flash Attention for SDPA backward; the current SYCL TLA
        implementation does not guarantee precision on PVC. Re-enable once oneDNN
        adds SDPA backward support. See https://github.com/intel/torch-xpu-ops/issues/4094"
```

**Review by guangyey — APPROVED** [review]

File: test/inductor/test_mix_order_reduction.py [hunk 2]
```diff
@@ -1161,6 +1161,11 @@ class OverFusionTest(TestBase):
     regression. See #179423.
     """
 
+    @skipIfXpu(
+        msg="XPU selects Flash Attention for SDPA backward; the current SYCL TLA
+        implementation does not guarantee precision on PVC. Re-enable once oneDNN
+        adds SDPA backward support. See https://github.com/intel/torch-xpu-ops/issues/4094"
```
Comment (guangyey) [inline test/inductor/test_mix_order_reduction.py]:
```suggestion
        msg="XPU selects Flash Attention for SDPA backward; the current SYCL TLA"
        "implementation does not guarantee precision on PVC. Re-enable once oneDNN"
        "adds SDPA backward support. See https://github.com/intel/torch-xpu-ops/issues/4094"
```

**Comment by ZhaoqiongZ** [comment]
Hi @jansel, all CI failures here are unrelated to this PR.
This PR only adds a `@skipIfXpu` decorator to one XPU unit test — it does not touch any CUDA code path or benchmark model. 

Could you take a look and merge when you get a chance? Thanks!

**Comment by jansel** [comment]
@claude explain those CI failures, are they related to this PR?

**Comment by ZhaoqiongZ** [comment]
Hi @jansel ,
Could you take a look at the analysis from claude and merge when you get a chance !

**Review by jansel — APPROVED** [review]
